linode/stackscript/framework_resource.go

```diff
@@ -0,0 +1,259 @@
+			"Unable to refresh the Linode StackScript",
+			fmt.Sprintf(
+				"Error finding the specified Linode StackScript: %s",
+				err.Error(),
+			),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(data.parseStackScript(ctx, stackscript)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *Resource) Update(
+	ctx context.Context,
+	req resource.UpdateRequest,
+	resp *resource.UpdateResponse,
+) {
+	var state StackScriptModel
+	var plan StackScriptModel
+
+	// Get the state & plan
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Get the ID from the plan
+	stackScriptID := int(helper.StringToInt64(state.ID.ValueString(), resp.Diagnostics))
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Check whether there were any changes
+	isUnchanged := state.Label.Equal(plan.Label) &&
+		state.Script.Equal(plan.Script) &&
+		state.Description.Equal(plan.Description) &&
+		state.RevNote.Equal(plan.RevNote) &&
+		state.IsPublic.Equal(plan.IsPublic) &&
+		state.Images.Equal(plan.Images)
+
+	// Apply the change if necessary
+	if !isUnchanged {
+		r.updateStackScript(ctx, resp, plan, stackScriptID)
+	}
+}
+
+func (r *Resource) Delete(
+	ctx context.Context,
+	req resource.DeleteRequest,
+	resp *resource.DeleteResponse,
+) {
+	var data StackScriptModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	stackscriptID := int(helper.StringToInt64(data.ID.ValueString(), resp.Diagnostics))
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	client := r.client
+	err := client.DeleteStackscript(ctx, stackscriptID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to delete the StackScript with id %v", stackscriptID),
+			err.Error(),
+		)
+		return
+	}
+}
+
+func (r *Resource) updateStackScript(
+	ctx context.Context,
+	resp *resource.UpdateResponse,
+	plan StackScriptModel,
+	stackScriptID int,
+) {
+	client := r.client
+
+	updateOpts := linodego.StackscriptUpdateOptions{
+		Label:       plan.Label.ValueString(),
+		Script:      plan.Script.ValueString(),
+		Description: plan.Description.ValueString(),
+		RevNote:     plan.RevNote.ValueString(),
+		IsPublic:    plan.IsPublic.ValueBool(),
+	}
+
+	// Special handling for images
+	resp.Diagnostics.Append(plan.Images.ElementsAs(ctx, &updateOpts.Images, false)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	stackScript, err := client.UpdateStackscript(ctx, stackScriptID, updateOpts)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Failed to update the StackScript with id %v", stackScriptID),
+			err.Error(),
+		)
+		return
+	}
+
+	resp.Diagnostics.Append(plan.parseStackScript(ctx, stackScript)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
```

Comment:
Optional suggestion: this line might be more suitable to be in the `Update` method?

Reply:
Never mind